src/Nethermind/Nethermind.Network/P2P/Subprotocols/Snap/SnapProtocolHandler.cs

```diff
@@ -0,0 +1,287 @@
+                    StorageRangeMessage storageRangesMessage = Deserialize<StorageRangeMessage>(message.Content);
+                    ReportIn(storageRangesMessage);
+                    Handle(storageRangesMessage, size);
+                    break;
+                case SnapMessageCode.GetByteCodes:
+                    GetByteCodesMessage getByteCodesMessage = Deserialize<GetByteCodesMessage>(message.Content);
+                    ReportIn(getByteCodesMessage);
+                    Handle(getByteCodesMessage);
+                    break;
+                case SnapMessageCode.ByteCodes:
+                    ByteCodesMessage byteCodesMessage = Deserialize<ByteCodesMessage>(message.Content);
+                    ReportIn(byteCodesMessage);
+                    Handle(byteCodesMessage, size);
+                    break;
+                case SnapMessageCode.GetTrieNodes:
+                    LogRequest(message.Content);
+                    GetTrieNodesMessage getTrieNodesMessage = Deserialize<GetTrieNodesMessage>(message.Content);
+                    ReportIn(getTrieNodesMessage);
+                    Handle(getTrieNodesMessage);
+                    break;
+                case SnapMessageCode.TrieNodes:
+                    TrieNodesMessage trieNodesMessage = Deserialize<TrieNodesMessage>(message.Content);
+                    ReportIn(trieNodesMessage);
+                    Handle(trieNodesMessage, size);
+                    break;
+            }
+        }
+
+        private void LogRequest(IByteBuffer buffer)
+        {
+            Logger.Info($"SNAP - GetNodeData:{Bytes.ToHexString(buffer.Array)}");
+        }
+
+        private void Handle(AccountRangeMessage msg, long size)
+        {
+            Metrics.SnapAccountRangeReceived++;
+            _getAccountRangeRequests.Handle(msg, size);
+        }
+
+        private void Handle(StorageRangeMessage msg, long size)
+        {
+            Metrics.SnapStorageRangesReceived++;
+            _getStorageRangeRequests.Handle(msg, size);
+        }
+
+        private void Handle(ByteCodesMessage msg, long size)
+        {
+            // TODO: increment metrics
+            _getByteCodesRequests.Handle(msg, size);
+        }
+
+        private void Handle(TrieNodesMessage msg, long size)
+        {
+            // TODO: increment metrics
+            _getTrieNodesRequests.Handle(msg, size);
+        }
+
+        private void Handle(GetAccountRangeMessage msg)
+        {
+            //throw new NotImplementedException();
+        }
+
+        private void Handle(GetStorageRangeMessage getStorageRangesMessage)
+        {
+            //throw new NotImplementedException();
+        }
+
+        private void Handle(GetByteCodesMessage getByteCodesMessage)
+        {
+            //throw new NotImplementedException();
+        }
+
+        private void Handle(GetTrieNodesMessage getTrieNodesMessage)
+        {
+            //throw new NotImplementedException();
+        }
```

Comment:
@marcindsobczak was looking into it.
An empty response in snap has some meaning (obsolete root hash) which in this case is not true but maybe it's better nothing.
It's not a big issue currently because we turn off snap sync after the sync finishes and in the meantime only other new syncing nodes send requests to us.

https://github.com/NethermindEth/nethermind/issues/4010